.github/workflows/lints.yml

```diff
@@ -0,0 +1,30 @@
+name: Lints
+on: [push, pull_request]
+
+jobs:
+  lints:
+    name: Lints
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: 1.59.0
+          override: true
+          components: rustfmt, clippy
+
+      - name: Run cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --all-targets -- -D warnings
+
+      - name: Run cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
```

Comment:
Clippy only displays lint warnings right? Can we set up rust-fmt so it fixes clippy lints?

Reply:
`cargo clippy --fix` only applies those lints that are marked as [MachineApplicable](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint_defs/enum.Applicability.html#variant.MachineApplicable), so it should be safe.